src/main/java/com/microsoft/azure/functions/worker/broker/JavaFunctionBroker.java

```diff
@@ -87,14 +93,48 @@ private void addSearchPathsToClassLoader(FunctionMethodDescriptor function) thro
 		function.getLibDirectory().ifPresent(d -> registerWithClassLoaderProvider(d));
 	}
 
-	private void registerWithClassLoaderProvider(File libDirectory) {
+	void registerWithClassLoaderProvider(File libDirectory) {
 		try {
-			classLoaderProvider.addDirectory(libDirectory);
+			if(SystemUtils.IS_JAVA_1_8) {
+				String workerLibPath = System.getenv(Constants.FUNCTIONS_WORKER_DIRECTORY) + "/lib";
+				String javaReverseLibLoading = System.getenv(Constants.FUNCTIONS_WORKER_JAVA_LOAD_APP_LIBS);
+				File workerLib = new File(workerLibPath);
+				if(Utilities.isTrue(javaReverseLibLoading)) {
+					if(verifyLibrariesExist (workerLib, workerLibPath, "CLIENT_LIB_FIRST")) {
```

Comment:
it should be enough to call `verifyLibrariesExist` once right? both calls use same arguments?

Reply:
The arguments are different but you raised good point we do not need to have the last param. I will make it once call.